org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/managers/MavenProjectImporterTest.java

```diff
@@ -363,6 +367,28 @@ public void avoidImportDuplicatedProjects() throws Exception {
 		}
 	}
 
+	// https://github.com/redhat-developer/vscode-java/issues/2712
+	@Test
+	public void testNullAnalysisDisabled2() throws Exception {
+		this.preferenceManager.getPreferences().setNonnullTypes(ImmutableList.of("javax.annotation.Nonnull", "org.eclipse.jdt.annotation.NonNull"));
+		this.preferenceManager.getPreferences().setNullableTypes(ImmutableList.of("org.eclipse.jdt.annotation.Nullable", "javax.annotation.Nonnull"));
+		try {
+			IProject project = importMavenProject("null-analysis");
+			assertIsJavaProject(project);
+			if (this.preferenceManager.getPreferences().updateAnnotationNullAnalysisOptions()) {
+				BuildWorkspaceHandler buildWorkspaceHandler = new BuildWorkspaceHandler(JavaLanguageServerPlugin.getProjectsManager());
+				buildWorkspaceHandler.buildWorkspace(true, new NullProgressMonitor());
+			}
+			IJavaProject javaProject = JavaCore.create(project);
+			Map<String, String> options = javaProject.getOptions(true);
+			assertEquals(options.get(JavaCore.COMPILER_ANNOTATION_NULL_ANALYSIS), JavaCore.DISABLED);
```

Comment:
reverse the arguments here. It's expected, actual.

Reply:
Fixed